obspy/io/css/core.py

```diff
@@ -134,11 +146,20 @@ def _read_css(filename, **kwargs):
         else:
             read_fmt = np.dtype(dtype)
             fmt = read_fmt
-        with open(filename, "rb") as fh:
-            fh.seek(offset)
-            data = fh.read(read_fmt.itemsize * npts)
-            data = from_buffer(data, dtype=read_fmt)
-            data = np.require(data, dtype=fmt)
+
+        try:
+            with open(dfilename, "rb") as fh:
+                data = _read_segment()
```

Comment:
This piece of code made me realize that I do not fully know how does the scope of variables work in Python. 

Anyway, I feel about not passing a arguments to a method directly and using variables from outside of its scope a bit unclear. But to be honest, I am not sure what is the obspy's policy on that.

Reply:
I rewrote it in a simple way to commit 0ab2244.

Reply:
I meant here to explicitly pass the fh as argument to read_segment but your solution also works fine :)